### PR TITLE
chore(eslint-config): shareable eslint config package

### PR DESCRIPTION
**Pull request by dpinol**
## Description

Eslint shareable configuration package.

## Context
Unifying the eslint configuration will simplify the eslint configuration of new botonic projects, specially those outside the "packages" folder (which at least inherit the eslint dependencies from the root folder). 

## Approach taken / Explain the design

As specified in the [official documentation](https://eslint.org/docs/developer-guide/shareable-configs)

## To document / Usage example

For a project to use this [eslint](https://eslint.org/) configuration, you'll need to:
* Add `extends: '@botonic/eslint-config/index.js',` to your .eslintrc.js
* If you can afford eslint to be slower (eg from your CI pipeline) to detect more issues, add '@botonic/eslint-config/index-ci.js'
* Add this script to your package.json: ` "lint": "eslint_d --fix --quiet 'src/**/*.js*' 'src/**/*.ts' "`.
* Execute `npm run lint`
Please check the [official user guide](https://eslint.org/docs/user-guide/) for adapting this configuration to your needs.

## Testing

It's being tested in projects outside the botonic repository.

### CODE REVIEW

**Review by vanbasten17 — APPROVED** [review]
LGTM! 👍  just one question, it will continue supporting projects which are not 100% TypeScript, right?

**Comment by dpinol** [comment]
> LGTM! +1 just one question, it will continue supporting projects which are not 100% TypeScript, right?

yes, and it will lint their .d.ts also :-) and even the types declared in JS files through jsdocs

File: packages/botonic-eslint-config/package.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    "bot-framework",
+    "eslint",
+    "eslintconfig",
+    "javascript",
+    "typescript"
+  ],
+  "dependencies": {
+    "@typescript-eslint/parser": "~4.19.0"
+  },
+  "peerDependencies": {
```
Comment (marcpdw) [inline packages/botonic-eslint-config/package.json:19]: Had never seen those. Are they complementing `dependencies` or what?

File: packages/botonic-eslint-config/index.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+      '@typescript-eslint/parser': ['.ts', '.tsx', '.d.ts'],
+    },
+    'import/resolver': {
+      typescript: {
+        alwaysTryTypes: true, // always try to resolve types under `<root>@types` directory even it doesn't contain any source code, like `@types/unist`
+      },
+    },
+  },
+  // npm run lint runs eslint with --quiet --fix so that only errors are fixed
+  rules: {
```
Comment (marcpdw) [inline packages/botonic-eslint-config/index.js:44]: Are the rules based on what you already had on Botonic or where are they coming from?

File: packages/botonic-eslint-config/index-ci.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+// Eslint configuration for CI (slower but detects more issues)
```
Comment (marcpdw) [inline packages/botonic-eslint-config/index-ci.js:1]: Does this mean that you might get issues that the editor is not telling you?

**Review by marcpdw — COMMENTED** [review]
I like this idea, good stuff putting it together!

It looks good overall, I just have a few questions if you don't mind.

File: packages/botonic-eslint-config/package.json (hunk 1 above)
Comment (dpinol) [reply packages/botonic-eslint-config/package.json:19]: https://nodejs.org/es/blog/npm/peer-dependencies/

File: packages/botonic-eslint-config/index.js (hunk 2 above)
Comment (dpinol) [reply packages/botonic-eslint-config/index.js:44]: yes, in botonic plugin contentful & dynamo

File: packages/botonic-eslint-config/index-ci.js (hunk 3 above)
Comment (dpinol) [reply packages/botonic-eslint-config/index-ci.js:1]: yes, It's a compromise between agility on development machines (these options make IDEs and github hooks very slow) and test thoroughness

**Review by marcpdw — APPROVED** [review]

**Pull request merged**